docs/src/lib/indexing.md

```diff
@@ -29,6 +29,12 @@ The rules for a valid type of index into a column are the following:
     * an `All` or `Between` expression (see [DataAPI.jl](https://github.com/JuliaData/DataAPI.jl));
     * a colon literal `:`.
 
+Indexing with `Not` has special behavior, in particular:
+* `String`s and `Symbol`s that do not belong in the column names of the data frame are ignored. The same applies for elements of `AbstractVector{<:String}` and `AbstractVector{Symbol}` that do not belong in the column names of the data frame; 
```

Comment:
```suggestion
* strings and `Symbol`s that do not belong in the column names of the data frame are ignored. The same applies for elements of `AbstractVector{<:AbstractString}` and `AbstractVector{Symbol}` that do not belong in the column names of the data frame; 
```

Reply:
I would also add that the same applies to the case if they are present in `All` expressions.

On the other hand - the three rules you described below are not special - they follow the standard rules for `Not` from InvertedIndices.jl. So we can do two things - either just remove them, or if you think it is worth to mention it, change the line above to be more verbose and say something like:
```
Indexing with `Not` selects indices that are a complement of the selection specified by its argument, in particular:
```